spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/data/neo4j/Neo4jDataAutoConfigurationTests.java

```diff
@@ -139,6 +142,41 @@ public void eventListenersAreAutoRegistered() {
 				.onPreSave(any(Event.class));
 	}
 
+	@Test
+	public void autoIndexConfigurationIsAssert() {
+		load(CustomConfigurationFactory.class, "spring.data.neo4j.indexes.auto=assert");
+		org.neo4j.ogm.config.Configuration configuration = this.context
+				.getBean(org.neo4j.ogm.config.Configuration.class);
+		assertThat(configuration.getAutoIndex()).isEqualTo(AutoIndexMode.ASSERT);
+	}
+
+	@Test
+	public void autoIndexConfigurationIsNone() {
+		load(CustomConfigurationFactory.class);
+		org.neo4j.ogm.config.Configuration configuration = this.context
+				.getBean(org.neo4j.ogm.config.Configuration.class);
+		assertThat(configuration.getAutoIndex()).isEqualTo(AutoIndexMode.NONE);
```

Comment:
There's not much use in having four tests that check that the four value are set in the right place.